common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -583,6 +686,7 @@ public MergeStatuses finalizeShuffleMerge(FinalizeShuffleMerge msg) {
         bitmaps.toArray(new RoaringBitmap[bitmaps.size()]), Ints.toArray(reduceIds),
         Longs.toArray(sizes));
     }
+    writeAppAttemptShuffleMergeInfoToDB(appAttemptShuffleMergeId);
```

Comment:
I think based on an earlier comment, this was moved here. However, I think we should write to db when we update the state that the partition is finalized. When a particular reduce partition fails to finalize, we don't add it to the result. However, we still mark the shuffle to be finalized. In that case we should update the database there as well.

Reply:
Does "Fails to finalize" mean this call `partition.finalizePartition();` fail? Even it fails to finalize the partition, with the current implementation, it will still write to DB. Probably I misunderstood the comment here. If you can elaborate more? Thanks.

Reply:
Updated.